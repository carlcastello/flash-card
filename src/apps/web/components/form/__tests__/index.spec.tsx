import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json'

import { Form }  from '../index';

describe("Web APP", () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(
      <Form
        fields={[
          {
            id: 'potato salad',
            label: 'potato-salad',
            type: 'string'
          }
        ]}
        isLoading={false}
        onSuccess={() => {}}
        classes={{
          formControl: 'form-control',
          formLabel: 'form-label',
          formInput: 'form-input',
          formHelperText: 'form-helper-text',
          formBottomContainer: 'form-bottom-container',
        }}/>
    );
  });


  it("Takes snapshot", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  })
});