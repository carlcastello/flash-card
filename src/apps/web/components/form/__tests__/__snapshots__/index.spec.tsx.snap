// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Web APP Takes snapshot 1`] = `
<form>
  <WithStyles(ForwardRef(FormControl))
    className="form-control"
    error={false}
    fullWidth={true}
    key="form-0"
    variant="outlined"
  >
    <WithStyles(ForwardRef(InputLabel))
      className="form-label"
      htmlFor="potato salad"
    >
      potato-salad
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(OutlinedInput))
      className="form-input"
      id="potato salad"
      label="potato-salad"
      onChange={[Function]}
    />
  </WithStyles(ForwardRef(FormControl))>
  <Styled(MuiBox)
    className="form-bottom-container"
  >
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
    >
      Save
    </WithStyles(ForwardRef(Button))>
  </Styled(MuiBox)>
</form>
`;
